docs/rfcs/component-universal-telemetry.md

```diff
@@ -0,0 +1,212 @@
+`success` otherwise.
+
+```yaml
+    otelcol.receiver.produced.items:
+      enabled: true
+      description: Number of items emitted from the receiver.
+      unit: "{items}"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.processor.consumed.items:
+      enabled: true
+      description: Number of items passed to the processor.
+      unit: "{items}"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.processor.produced.items:
+      enabled: true
+      description: Number of items emitted from the processor.
+      unit: "{items}"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.connector.consumed.items:
+      enabled: true
+      description: Number of items passed to the connector.
+      unit: "{items}"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.connector.produced.items:
+      enabled: true
+      description: Number of items emitted from the connector.
+      unit: "{items}"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.exporter.consumed.items:
+      enabled: true
+      description: Number of items passed to the exporter.
+      unit: "{items}"
+      sum:
+        value_type: int
+        monotonic: true
+
+   otelcol.receiver.produced.size:
+      enabled: false
+      description: Size of items emitted from the receiver.
+      unit: "By"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.processor.consumed.size:
+      enabled: false
+      description: Size of items passed to the processor.
+      unit: "By"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.processor.produced.size:
+      enabled: false
+      description: Size of items emitted from the processor.
+      unit: "By"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.connector.consumed.size:
+      enabled: false
+      description: Size of items passed to the connector.
+      unit: "By"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.connector.produced.size:
+      enabled: false
+      description: Size of items emitted from the connector.
+      unit: "By"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.exporter.consumed.size:
+      enabled: false
+      description: Size of items passed to the exporter.
+      unit: "By"
+      sum:
+        value_type: int
+        monotonic: true
+```
+
+### Auto-Instrumented Logs
+
+Metrics provide most of the observability we need but there are some gaps which logs can fill. Although metrics would describe the overall
+item counts, it is helpful in some cases to record more granular events. e.g. If a produced batch of 10,000 spans results in an error, but
+100 batches of 100 spans succeed, this may be a matter of batch size that can be detected by analyzing logs, while the corresponding metric
```

Comment:
Makes sense if we agree that we should capture a span for the consume function.

Reply:
It would be nice to have a clear consensus on this issue of logs vs. spans for information not easily captured by metrics, so I can try to align with that for #11581.

Reply:
I've opened https://github.com/open-telemetry/opentelemetry-collector/issues/11743 to track this discussion